File: examples/python/consumer.py
import os
from confluent_kafka import Consumer

BOOTSTRAP = os.environ["KAFKA_BOOTSTRAP"]
TOPIC = os.environ.get("KAFKA_TOPIC", "app1.events")
USERNAME = os.environ["KAFKA_USERNAME"]
PASSWORD = os.environ["KAFKA_PASSWORD"]
CA_LOCATION = os.environ.get("KAFKA_CA_LOCATION", "./ca.crt")
GROUP = os.environ.get("KAFKA_GROUP", "app1")

consumer = Consumer(
    {
        "bootstrap.servers": BOOTSTRAP,
        "security.protocol": "SASL_SSL",
        "sasl.mechanisms": "SCRAM-SHA-512",
        "sasl.username": USERNAME,
        "sasl.password": PASSWORD,
        "ssl.ca.location": CA_LOCATION,
        "group.id": GROUP,
        "auto.offset.reset": "earliest",
    }
)

consumer.subscribe([TOPIC])
print(f"Consuming {TOPIC}...")

try:
    while True:
        msg = consumer.poll(1.0)
        if msg is None:
            continue
        if msg.error():
            print(f"error: {msg.error()}")
            continue
        print(f"{msg.topic()}[{msg.partition()}]@{msg.offset()}: {msg.value().decode('utf-8')}")
finally:
    consumer.close()

File: examples/node/consumer.mjs
import fs from 'node:fs';
import { Kafka } from 'kafkajs';

const bootstrap = process.env.KAFKA_BOOTSTRAP;
const topic = process.env.KAFKA_TOPIC ?? 'app1.events';
const username = process.env.KAFKA_USERNAME;
const password = process.env.KAFKA_PASSWORD;
const caPath = process.env.KAFKA_CA_LOCATION ?? './ca.crt';
const groupId = process.env.KAFKA_GROUP ?? 'app1';

if (!bootstrap || !username || !password) {
    throw new Error('Missing env: KAFKA_BOOTSTRAP, KAFKA_USERNAME, KAFKA_PASSWORD');
}

const kafka = new Kafka({
    clientId: 'node-consumer',
    brokers: [bootstrap],
    ssl: { ca: [fs.readFileSync(caPath, 'utf8')] },
    sasl: { mechanism: 'scram-sha-512', username, password },
});

const consumer = kafka.consumer({ groupId });
await consumer.connect();
await consumer.subscribe({ topic, fromBeginning: true });

await consumer.run({
    eachMessage: async ({ topic, partition, message }) => {
        console.log(`${topic}[${partition}]@${message.offset}: ${message.value?.toString()}`);
    },
});
